tests/integration/cqlengine/columns/test_validation.py

```diff
@@ -187,49 +184,173 @@ def test_varint_io(self):
         int2 = self.VarIntTest.objects(test_id=0).first()
         self.assertEqual(int1.bignum, int2.bignum)
 
+        with self.assertRaises(ValidationError):
+            self.VarIntTest.objects.create(test_id=0, bignum="not_a_number")
 
-class TestDate(BaseCassEngTestCase):
-    class DateTest(Model):
-
-        test_id = Integer(primary_key=True)
-        created_at = Date()
 
+class DataType():
     @classmethod
     def setUpClass(cls):
-        if PROTOCOL_VERSION < 4:
+        if PROTOCOL_VERSION < 4 or CASSANDRA_VERSION < "3.0":
             return
-        sync_table(cls.DateTest)
+
+        class DataTypeTest(Model):
+            test_id = Integer(primary_key=True)
+            class_param = cls.db_klass()
+
+        cls.model_class = DataTypeTest
+        sync_table(cls.model_class)
 
     @classmethod
     def tearDownClass(cls):
-        if PROTOCOL_VERSION < 4:
+        if PROTOCOL_VERSION < 4 or CASSANDRA_VERSION < "3.0":
```

Comment:
Otherwise we would get the error:
`InvalidRequest: Error from server: code=2200 [Invalid query] message="Predicates on non-primary-key columns (created_at) are not yet supported for non secondary index queries"` for versions 2.X